src/data/real/sqrt.lean

```diff
@@ -196,6 +211,26 @@ begin
   { exact le_trans hx (sqrt_nonneg y) }
 end
 
+lemma lt_sqrt (hx : 0 ≤ x) (hy : 0 ≤ y) : x < sqrt y ↔ x ^ 2 < y :=
+by rw [mul_self_lt_mul_self_iff hx (sqrt_nonneg y), pow_two, mul_self_sqrt hy]
+
+lemma lt_sqrt_of_sqr_lt (h : x^2 < y) : x < sqrt y :=
+begin
+  by_contra hnot,
+  rw [le_antisymm (le_sqrt_of_sqr_le _) (not_lt.mp hnot), sqr_sqrt] at h,
+  exacts [h.false, (lt_of_le_of_lt (pow_two_nonneg _) h).le, h.le],
+end
+
+lemma sqr_le (h : x^2 ≤ y) : -sqrt y ≤ x ∧ x ≤ sqrt y :=
+abs_le.mp (by simpa [← sqrt_sqr_eq_abs] using sqrt_le_sqrt h)
+
+lemma sqr_le_of_nonneg (h : 0 ≤ y) : x^2 ≤ y ↔ -sqrt y ≤ x ∧ x ≤ sqrt y :=
+⟨sqr_le, (by rw [← abs_le, ← sqr_abs]; exact (le_sqrt (abs_nonneg x) h).mp)⟩
+
+lemma sqr_le_left (h : x^2 ≤ y) : -sqrt y ≤ x := (sqr_le h).1
+
+lemma sqr_le_right (h : x^2 ≤ y) : x ≤ sqrt y := (sqr_le h).2
```

Comment:
These lemma names seem odd to me, as they don't say anything about their conclusion